File: robot/src/main/java/frc/team254/lib/trajectory/TrajectoryView.java
package frc.team254.lib.trajectory;

import frc.team5190.lib.geometry.interfaces.State;

public interface TrajectoryView<S extends State<S>> {
    public TrajectorySamplePoint<S> sample(final double interpolant);

    public double first_interpolant();

    public double last_interpolant();

    public Trajectory<S> trajectory();
}

File: robot/src/main/kotlin/frc/team5190/lib/trajectory/TrajectoryView.kt
/*
 * FRC Team 5190
 * Green Hope Falcons
 */

/*
 * Some implementations and algorithms borrowed from:
 * NASA Ames Robotics "The Cheesy Poofs"
 * Team 254
 */


package frc.team5190.lib.trajectory

import frc.team5190.lib.geometry.interfaces.State

interface TrajectoryView<S : State<S>> {

    fun sample(interpolant: Double): TrajectorySamplePoint<S>

    val firstInterpolant: Double
    val lastInterpolant: Double

    val trajectory: Trajectory<S>
}
